docs/guides/color.md

```diff
@@ -0,0 +1,290 @@
+    [0, 0, 1],
+])
+points.face_color = np.array([
+    [1, 0, 0, 1],
+    [0, 1, 0, 1],
+    [0, 0, 1, 1],
+])
+```
+
+However, when specifying many colors in a sequence or generator *do not mix different single color types*.
+For example, something like
+
+```python
+In [5]: points.face_color = np.array(['red', (0, 1, 0), '#0000ff'])
+```
+
+is not supported or guaranteed to work.
+
+
+## Colormaps
+
+In general, a colormap is a function that maps from an arbitrary domain to a some color space range.
+As napari's standard form of color is an RGBA array, this range is the RGBA color space.
+
+In napari, colormaps are used in two ways.
+- To map an image pixel value to an RGBA color.
+- To map a property value to an RGBA color.
+
+In both cases the types of values in the domain may be continuous (e.g. `float`)
+or categorical (e.g. `int`, `str`), so napari has two different types of colormaps to handle those cases.
+
+
+### Continuous maps
+
+The [`Colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap.py#L28)
+class defines a map from a continuous or real value to a color.
+For example, a 2D image of floats between 0 and 1 can be mapped to colors.
+
+```python
+In [6]: image.colormap = 'blue'
+```
+
+Internally, the [`ensure_colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap_utils.py#L492)
+function is used to coerce colormap input values to an instance of napari's `Colormap`.
+
+```python
+In [7]: image.colormap
+Out[7]:
+Colormap(
+    colors=np.array([0., 0., 0., 1], [0., 0., 1., 1]]),
+    name='blue',
+    interpolation=<ColormapInterpolationMode.LINEAR: 'linear'>,
+    controls=array([0., 1.]),
+)
+```
+
+In this case, values of 0 are mapped to black (`[0, 0, 0, 1]`),
+values of 1 are mapped to blue (`[0, 0, 1, 1]`),
+and values in `(0, 1)` are mapped to colors that are linearly interpolated
+between black and blue.
+
+A few types of input are supported.
+The easiest to use is the colormap name as above.
+```python
+image.colormap = 'blue'
+```
+
+By default colormap names come from one of the following sources.
+
+- Some of [matplotlib's colormap names](https://matplotlib.org/stable/tutorials/colors/colormaps.html).
+- Standard optical primary and secondary colors: `'red', 'green', 'blue', 'yellow', 'magenta', 'cyan'`.
+- BOP (blue, orange, purple) colormap names: `'bop blue', 'bop orange', 'bop purple'`.
+
+More specifically, the available colormap names can be listed at any time.
+
+```python
+In [8]: list(napari.utils.colormaps.AVAILABLE_COLORMAPS)
+Out[8]:
+['PiYG',
+ 'blue',
+ 'bop blue',
+ ...,
+ 'viridis',
+ 'yellow']
+```
+
+If there is a need for a custom colormap that is not available by default,
+pass an instance of a napari `Colormap`
+
+
+```python
+image.colormap = Colormap(
+    colors=[[0, 0, 0, 1], [0, 0, 1, 1]],
+    name='blue',
+    interpolation='linear',
+    controls=[0, 1],
+)
```

Comment:
i think it would be nice to add the `Colormap` import to this snippet.

Reply:
Done.